zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageSubscriptionCorrelateProcessor.java

```diff
@@ -71,10 +76,39 @@ public void processRecord(final TypedRecord<MessageSubscriptionRecord> record) {
     if (subscription == null) {
       rejectCommand(record);
       return;
+
+    } else if (subscription.getRecord().getMessageKey() != record.getValue().getMessageKey()) {
```

Comment:
💭 In theory it could happen we already have this command on the log where the message key will still be `-1` as the default. We will reject the command and not do anything. The message correlation will then be retried and that acknowledgement will get a message key. The other partition will detect it already correlated and just send the acknowledgement without correlating it again.

Just dumping this thought for myself to see if it'd work. It looks like it does, but there's a chance we run into this scenario. In the end it results in an extra back and forth between partitions but that should be fine imo.